### PR TITLE
Add page visibility state functionality

### PR DESCRIPTION
**Pull request by garyb**
**Description of the change**

Adds type and property for page visibility state, as per https://html.spec.whatwg.org/multipage/interaction.html#page-visibility

Also did a little housekeeping while I was in there.

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: src/Web/HTML/HTMLDocument/VisibilityState.purs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+instance showVisibilityState :: Show VisibilityState where
+  show = case _ of
+    Visible -> "Visible"
+    Hidden -> "Hidden"
+
+print :: VisibilityState -> String
+print = case _ of
+  Visible -> "loading"
+  Hidden -> "interactive"
```
Comment (JordanMartinez) [inline src/Web/HTML/HTMLDocument/VisibilityState.purs]: Could you clarify where this is stated in the doc you referenced?

**Comment by JordanMartinez** [comment]
The other commits otherwise look fine.

File: src/Web/HTML/HTMLDocument/VisibilityState.purs (hunk 1 above)
Comment (thomashoneyman) [reply src/Web/HTML/HTMLDocument/VisibilityState.purs]: Yeah, shouldn’t this be “hidden” and “visible”? Reading through the spec I didn’t see mention of these wrt visibility state.

Comment (garyb) [reply src/Web/HTML/HTMLDocument/VisibilityState.purs]: Ahem uhm err, I definitely didn't copy and paste this from `ReadyState` and forget to update these 🤦‍♂️

**Review by JordanMartinez — APPROVED** [review]

**Comment by garyb** [comment]
Thanks for taking a look and spotting that! ❤️

**Pull request merged**